src/puppetlabs/ezbake/core.clj

```diff
@@ -501,23 +541,112 @@ Dependency tree:
 
 (defn generate-project-data-yaml
-  [lein-project build-target]
+  [lein-project build-target additional-uberjars]
   (lein-main/info "generating project_data.yaml file")
   (spit
     (fs/file staging-dir "ext" "project_data.yaml")
     (stencil/render-string
       (slurp (get-staging-template-file "project_data.yaml.mustache"))
-      {:project       (:name lein-project)
-       :summary       (:description lein-project)
-       :description   (format "%s (%s)"
-                              (:description lein-project)
-                              (deputils/generate-manifest-string lein-project))
-       :uberjar-name  (:uberjar-name lein-project)
-       :is-pe-build   (format "%s" (= (get-local-ezbake-var lein-project :build-type "foss") "pe"))
-       :repo-name     (format "%s" (get-local-ezbake-var lein-project :repo-target ""))})))
+      {:project (:name lein-project)
+       :summary (:description lein-project)
+       :description (format "%s (%s)"
+                            (:description lein-project)
+                            (deputils/generate-manifest-string lein-project))
+       :uberjar-name (:uberjar-name lein-project)
+       :additional-uberjars (mapv (fn [filename] {:uberjar filename}) additional-uberjars)
+       :is-pe-build (format "%s" (= (get-local-ezbake-var lein-project :build-type "foss") "pe"))
+       :repo-name (format "%s" (get-local-ezbake-var lein-project :repo-target ""))})))
+
+(schema/defn get-additional-uberjars
+  "Returns the list of additional uberjar dependencies from the given lein project"
+  [lein-project]
+  (when-let [dependencies-vector (get-in lein-project [:lein-ezbake :additional-uberjars])]
+    dependencies-vector))
+
+(schema/defn resolve-dependency! :- File
+  "Resolves a single dependency and returns a File pointing to it's jar.
+
+  This has the side effect of fetching the whole dependency tree for the given
+  dependency, but we only care about the one jar"
+  [[project-symbol version :as dependency-coordinates]
+   repositories]
+  (lein-main/info "Resolving dependency for " dependency-coordinates)
+  (let [project {:dependencies [dependency-coordinates]
+                 :repositories repositories}
+        ; resolved-dependencies will be a list of jars, one for each dependency
+        ; in the project, including the jar for this project
+        resolved-dependencies (lein-classpath/resolve-managed-dependencies :dependencies nil project)
+        jar-regex (re-pattern (format "%s-%s.jar" (name project-symbol) version))
+        jar-file (first (filter #(re-find jar-regex (.getName %))
+                                resolved-dependencies))]
+    jar-file))
+
+(schema/defn extract-project-from-jar! :- schema/Str
+  "Extracts the project.clj file out of the given jar into the destination directory"
+  [jar :- File
+   destination-dir :- schema/Str]
+  (fs/mkdirs destination-dir)
+  (let [jar-file (JarFile. jar)
+        jar-entry (.getJarEntry jar-file "project.clj")
```

Comment:
Cool, this makes sense. I'm alright with how it is now, and I think so long as the docs mention that it's using the `project.clj` (which they are), then this is all good.